nixos/modules/services/web-servers/phpfpm/default.nix

```diff
@@ -133,14 +128,16 @@ let
       };
 
       config = {
-        socket = if poolOpts.listen == "" then "${runtimeDir}/${name}.sock" else poolOpts.listen;
         group = mkDefault poolOpts.user;
         phpOptions = mkBefore cfg.phpOptions;
 
-        settings = mapAttrs (name: mkDefault){
-          listen = poolOpts.socket;
-          user = poolOpts.user;
-          group = poolOpts.group;
+        settings = mapAttrs (name: mkDefault) {
+          "listen" = poolOpts.socket;
+          "listen.owner" = poolOpts.user;
+          "listen.group" = poolOpts.group;
+          "listen.mode" = "0660";
+          "user" = poolOpts.user;
+          "group" = poolOpts.group;
```

Comment:
After delete these lines - Service stops starting.
Worked with this lines:
```
          RuntimeDirectory = "phpfpm-{pool}";
          RuntimeDirectoryMode = "0750";
```
With `RuntimeDirectory = "phpfpm";` and 2 pools the directory `/run/phpfpm` is recreated. Need use `"phpfpm-{pool}";`

Reply:
my example patch
```
diff --git a/nixos/modules/services/web-servers/phpfpm/default.nix b/nixos/modules/services/web-servers/phpfpm/default.nix
index 31f4ae5445f..9e0177c22c8 100644
--- a/nixos/modules/services/web-servers/phpfpm/default.nix
+++ b/nixos/modules/services/web-servers/phpfpm/default.nix
@@ -5,8 +5,6 @@ with lib;
 let
   cfg = config.services.phpfpm;

-  runtimeDir = "/run/phpfpm";
-
   toStr = value:
     if true == value then "yes"
     else if false == value then "no"
@@ -42,7 +40,7 @@ let
       options = {
         socket = mkOption {
           type = types.str;
-          default = "${runtimeDir}/${name}.sock";
+          default = "/run/phpfpm-${name}/${name}.sock";
           readOnly = true;
           description = ''
             Path to the unix socket file on which to accept FastCGI requests.
@@ -136,8 +134,6 @@ let
           "listen.owner" = poolOpts.user;
           "listen.group" = poolOpts.group;
           "listen.mode" = "0660";
-          "user" = poolOpts.user;
-          "group" = poolOpts.group;
         };
       };
     };
@@ -147,6 +143,7 @@ in {
     (mkRemovedOptionModule [ "services" "phpfpm" "poolConfigs" ] "Use services.phpfpm.pools instead.")
     (mkRemovedOptionModule [ "services" "phpfpm" "phpIni" ] "")
     (mkRemovedOptionModule [ "services" "phpfpm" "listen" ] "Reference the read-only option config.services.phpfpm.pools.<name>.socket to access the path of your socket.")
+  ];

   options = {
     services.phpfpm = {
@@ -329,6 +326,8 @@ in {
           KillMode = mkIf createSocket "process";
           User = poolOpts.user;
           Group = poolOpts.group;
+          RuntimeDirectory = "phpfpm-${pool}";
+          RuntimeDirectoryMode = "0750";
         };
       }
     ) cfg.pools;

```

Reply:
`listen.owner`, `listen.group`, `listen.mode`, `user`, and `group` are all used by the nixos module and will not be used by `phpfpm` anymore. The reason they remain is documentation. If those values aren't there users can be confused what is happening. Tricky because `systemd` is taking care of everything that `phpfpm` used to... but better than leaving users without any idea IMO.

Reply:
`RuntimeDirectory` shouldn't be needed anymore because `systemd` `socket` is taking care of directory and socket creation.

Reply:
Options `user` and `group` writed to phpfpm configuration:
`cat /nix/store/...-phpfpm-test-01.conf`
```
[global]
daemonize = no
error_log = syslog
rlimit_core = unlimited
rlimit_files = 131072


[test-01]
group = nginx
listen = /run/phpfpm-test-01/test-01.sock
listen.group = nginx
listen.mode = 0660
listen.owner = nginx
pm = dynamic
pm.max_children = 20
pm.max_requests = 400
pm.max_spare_servers = 12
pm.min_spare_servers = 2
pm.start_servers = 4
user = nginx
```